src/transformers/visitors/circuitInternalFunctionCallVisitor.ts

```diff
@@ -19,8 +20,13 @@ const reorderParameters = (parameterList: any) => {
           parameterList[index] = newParam;
         }
       }
+      if(param.name === newParam.name && param.nodeType === 'VariableDeclaration' && newParam.nodeType === 'Boilerplate'){
+        !deletedIndexes.includes(index)? deletedIndexes.push(index) : deletedIndexes;
+      }
+             
     });
   });
+  deletedIndexes.forEach(index => parameterList.splice(index, 1));
```

Comment:
We have to be careful here as well. Say we have indexes (2,5) stored in deleted indexes, and we delete index 2 first then the element from index 5 is now in index 4 so the wrong index gets deleted. We need to go through the indexes in reverse order from highest to lowest.